neonvm/runner/main.go

```diff
@@ -359,19 +417,183 @@ func calcDirUsage(dirPath string) (int64, error) {
+	)
+
+	if err := execFg(QEMU_IMG_BIN, "create", "-q", "-f", "raw", "swap.raw", fmt.Sprintf("%d", totalDiskSize)); err != nil {
+		return err
+	}
+
+	if err := execFg("parted", "swap.raw", "mktable", "GPT"); err != nil {
+		return err
+	}
+	swapEndLocation := swapStartLocation + partitionSize
+	err := execFg(
+		"parted",
+		"--align=optimal",
+		"swap.raw",
+		"mkpart",
+		"primary",
+		// note the 'B' suffix, indicating units of bytes.
+		fmt.Sprintf("%dB", swapStartLocation),
+		fmt.Sprintf("%dB", swapEndLocation),
+	)
+	if err != nil {
+		return err
+	}
+
+	// Fetch the sector size, so we can pass the info to createISO9660runtime's logic for
+	// resize-swap.sh. `parted`'s output is difficult to parse, so we're using `sfdisk` instead.
+	// And, because `sfdisk` supports a `--json` output option, we can just directly deserialize
+	// from that to make our lives easier.
+	//
+	// Here, we only define the fields we actually care about. The actual JSON format looks
+	// something like:
+	//
+	//   {
+	//      "partitiontable": {
+	//         "label": "gpt",
+	//         "id": "52894F14-7333-4F50-B64C-3E7E2B3DCF7A",
+	//         "device": "swap.raw",
+	//         "unit": "sectors",
+	//         "firstlba": 34,
+	//         "lastlba": 6291422,
+	//         "sectorsize": 512,
+	//         "partitions": [
+	//            {
+	//               "node": "swap.raw1",
+	//               "start": 4096,
+	//               "size": 6285312,
+	//               "type": "0FC63DAF-8483-4772-8E79-3D69D8477DE4",
+	//               "uuid": "3E7FC783-06CC-474C-9E1E-5DD4A1E379A8",
+	//               "name": "primary"
+	//            }
+	//         ]
+	//      }
+	//   }
+	type sfdiskPartitionTable struct {
+		// nb: use a pointer so we can tell if it's not present
+		SectorSize *int64 `json:"sectorsize"`
+	}
+	type sfdiskOutput struct {
+		PartitionTable sfdiskPartitionTable `json:"partitiontable"`
+	}
+	sfdiskDump, err := execFgCapture("sfdisk", "--dump", "--json", "swap.raw")
+	if err != nil {
+		return err
+	}
+	logger.Info("Got swap file partition dump", zap.String("sfdiskOutput", string(sfdiskDump)))
+	var sfdiskOut sfdiskOutput
+	if err := json.Unmarshal(sfdiskDump, &sfdiskOut); err != nil {
+		return fmt.Errorf("failed to unmarshal sfdisk output: %w", err)
+	}
+	if sfdiskOut.PartitionTable.SectorSize == nil {
+		return errors.New("sfdisk output missing sector size")
+	}
+	// Now that we have sector size, communicate that info to createISO9660runtime, which needs it
+	// to create resize-swap.sh:
+	swapParamsResult <- swapParams{
+		sectorSize:            *sfdiskOut.PartitionTable.SectorSize,
+		partitionStartInBytes: swapStartLocation,
 	}
 
-	if err := execFg(QEMU_IMG_BIN, "create", "-q", "-f", "raw", "swap.raw", fmt.Sprintf("%d", diskSize.Value())); err != nil {
+	// Create the swap. We'll always resize it from within the guest, so we just need it to provide
+	// the label for blkid. Technically we only need to copy the first page (that's where the header
```

Comment:
Did you consider using PARTLABEL instead of LABEL?

This would mean using `sudo blkid -t PARTLABEL=main  -o device` to match and `parted name NUMBER NAME` to rename. But no need for mkswap on the host.